File: calculator/Calculator.fs
module Calculator

open Keyboard
open Cpu
open Display

type Calculator (cpu: Cpu, keyboard: Keyboard, display: Display) as this =

    do
        cpu.SetNumberHandler (Some display.Add)
        cpu.SetDecimalHandler (Some display.SetDecimal)
        cpu.SetErrorHandler (Some display.SetError)
        cpu.SetSignalHandler (Some display.SetSignal)
        cpu.SetClearHandler (Some display.Clear)
        keyboard.SetOperationHandler (Some cpu.Process)

    member public _.Keyboard
        with get () = this.Keyboard

    member public _.Display
        with get () = this.Display

File: cwlculator.cs/Calculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Calculator;
public class Calculator
{
    private readonly ICpu cpu;

    public IDisplay Display { get; }
    public IKeyboard Keyboard { get; }

    public Calculator(IDisplay iDisplay, IKeyboard iKeyboard, ICpu iCpu)
    {
        Display = iDisplay;
        Keyboard = iKeyboard;

        cpu = iCpu;

        cpu.ClearEvent += Display.Clear;
        cpu.DataEvent += Display.AddNumber;
        cpu.DecimalEvent += Display.SetDecimal;
        cpu.ErrorEvent += Display.SetError;
        cpu.NegativeEvent += Display.SetNegative;
        cpu.NumberEvent += Display.Add;

        Keyboard.KeyHandlerEvent += cpu.Process;
    }
}
